yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -322,8 +322,10 @@ private[spark] class Client(
     // multiple times, YARN will fail to launch containers for the app with an internal
     // error.
     val distributedUris = new HashSet[String]
-    obtainTokenForHiveMetastore(sparkConf, hadoopConf, credentials)
-    obtainTokenForHBase(sparkConf, hadoopConf, credentials)
+    if (isClusterMode) {
```

Comment:
why is this cluster mode only?  I can run spark shell to access hive or hbase and this won't get tokens for those to ship to executors?